### PR TITLE
K8SPSMDB-387 feature: add no oplogs check for PITRestore

### PR DESCRIPTION
**Pull request by ollevche**
[K8SPSMDB-387](https://jira.percona.com/browse/K8SPSMDB-387) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPSMDB-387

### CODE REVIEW

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| demand-backup-sharded | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| upgrade | passed |
| liveness | passed |
| upgrade-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/628/commits/70d68af7ca6dab179f6f4295fadf2a3907dc893d
image: `perconalab/percona-server-mongodb-operator:PR-628-70d68af7`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| arbiter | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| upgrade | passed |
| liveness | passed |
| upgrade-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | failed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/628/commits/9b6a959811c52c143e8662dbc3bc6d82704f6d6c
image: `perconalab/percona-server-mongodb-operator:PR-628-9b6a9598`

File: pkg/psmdb/backup/pbm.go [hunk 1]
```diff
@@ -209,3 +210,12 @@ func (b *PBM) GetLastPITRChunk() (*pbm.PITRChunk, error) {
...
 	return b.C.PITRLastChunkMeta(nodeInfo.SetName)
 }
+
+func (b *PBM) GetPITRChunkContains(unixTS int64) (*pbm.PITRChunk, error) {
+	nodeInfo, err := b.C.GetNodeInfo()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting node information")
+	}
+
+	return b.C.PITRGetChunkContains(nodeInfo.SetName, primitive.Timestamp{T: uint32(unixTS)})
```
Comment (qJkee) [inline pkg/psmdb/backup/pbm.go:220]: i think it's a little better to return `ErrNoOplogsForPITR` from there, instead of using tonn of `if` in controller

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/628/commits/9b6a959811c52c143e8662dbc3bc6d82704f6d6c
image: `perconalab/percona-server-mongodb-operator:PR-628-9b6a9598`

File: pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go [hunk 2]
```diff
@@ -267,6 +268,8 @@ func reEnablePITR(pbm *backup.PBM, backup psmdbv1.BackupSpec) (err error) {
 	return
 }
 
+var ErrNoOplogsForPITR = errors.New("There is no oplogs that can cover the date/time or no oplogs at all")
```
Comment (blez) [inline pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go:271]: 1. no need to export it.
2. errors should be in lower case
3. no need to do it global

File: pkg/psmdb/backup/pbm.go (hunk 1 above)
Comment (ollevche) [reply pkg/psmdb/backup/pbm.go]: changed in e03b30d

File: pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go (hunk 2 above)
Comment (ollevche) [reply pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go]: changed in e03b30d (but the err is still global due to its usage in different funcs)

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| security-context | passed |
| smart-update | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/628/commits/e03b30d63ea8664e30187cd3fd6a97e3f508bec6
image: `perconalab/percona-server-mongodb-operator:PR-628-e03b30d6`

**Review by blez — APPROVED** [review]

**Pull request merged**